### PR TITLE
[GPU] BugFix reduce_b_fs_yx_fsv16 kernel

### PR DESCRIPTION
**Pull request by byungilm**
+ Invalid calculation in reducing un-aligned feature axis for b_fs_yx_fsv16

### Details:
 - In reducing un-aligned feature axis, reduce calculation uses the whole 16 feature loaded by sub_group_block_read
 - Enabled unit-tests of reduce_eltwise_activation_quantize.basic on dGPU (blocked by https://github.com/openvinotoolkit/openvino/pull/15873)

### Tickets:
 - 67491

### CODE REVIEW

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl [hunk 1]
```diff
@@ -336,21 +336,17 @@ uint offset = batch_out * input_batch_pitch + ((feature_out + FSV - 1) / FSV) *
         for (uint fi = feature_out; fi < feature_max_val; fi += FSV) {
             for (uint yi = y_out; yi < y_max_val; ++yi) {
                 for (uint xi = x_out; xi < x_max_val; ++xi) {
-#if HANDLE_FEATURE_REMAINDER
```
Comment (vladimir-paramuzov) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl:339]: Please check the performance for this patch

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl [hunk 2]
```diff
@@ -330,27 +330,29 @@ uint offset = batch_out * input_batch_pitch + ((feature_out + FSV - 1) / FSV) *
...
-#else
+#else  // !REDUCE_X
     ACCUMULATOR_VEC acc = (ACCUMULATOR_VEC)(INIT_VAL);
     for (uint bi = batch_out; bi < batch_max_val; ++bi) {
         for (uint fi = feature_out; fi < feature_max_val; fi += FSV) {
             for (uint yi = y_out; yi < y_max_val; ++yi) {
                 for (uint xi = x_out; xi < x_max_val; ++xi) {
-#if HANDLE_FEATURE_REMAINDER
-                    INPUT_VEC input = (INPUT_VEC)(INPUT_INIT_VAL);
                     #if REDUCE_FEATURE && (INPUT0_FEATURE_NUM % FSV != 0)
```
Comment (isanghao) [inline src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl:339]: what about:
`#if REDUCE_FEATURE && (INPUT0_FEATURE_NUM % FSV != 0) && !ZERO_INVARIANT_REDUCTION`

File: src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp [hunk 3]
```diff
@@ -34,9 +34,13 @@ struct reduce_test_params {
 
 class ReduceFusingTest : public ::BaseFusingTest<reduce_test_params> {
 public:
-    void execute(reduce_test_params& p, bool is_dynamic = false) {
+    void execute(reduce_test_params& p, bool is_dynamic = false, bool use_default_input = false) {
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp:37]: what about `use_planar_input` instead?

File: src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp [hunk 4]
```diff
@@ -67,6 +71,10 @@ class ReduceFusingTest : public ::BaseFusingTest<reduce_test_params> {
         return layout{ p.in_shape, p.data_type, p.input_format };
     }
 
+    layout get_default_format_input_layout(reduce_test_params& p) {
+        return layout{ p.in_shape, p.data_type, format::bfyx };
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp:75]: to make it future proof, I think we should use `get_default_format(p.input_format)`

File: src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp [hunk 5]
```diff
@@ -134,28 +143,29 @@ TEST_P(reduce_eltwise_activation_quantize, basic) {
     );
 
     tolerance = 1.f;
-    execute(p);
+    execute(p, false, true);
```
Comment (isanghao) [inline src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp:146]: could you add a comment why this change is necessary?

**Review by isanghao — CHANGES_REQUESTED** [review]

File: src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp (hunk 4 above)
Comment (byungilm) [reply src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp:75]: I added a flag to get and to use default format in get_input_layout. How's it?

File: src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp (hunk 5 above)
Comment (byungilm) [reply src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp:146]: I added an explanation.

File: src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp (hunk 3 above)
Comment (byungilm) [reply src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp:37]: How about adding comment why use default format for using planar input here?
4 or 5 dims test-cases could be added later.

File: src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp (hunk 4 above)
Comment (byungilm) [reply src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp:75]: After short discussion about 'default format', here it uses default formats given by format::get_default_format.

File: src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp (hunk 3 above)
Comment (byungilm) [reply src/plugins/intel_gpu/tests/unit/fusions/reduce_fusion_test.cpp:37]: Updated.

File: src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl (hunk 2 above)
Comment (byungilm) [reply src/plugins/intel_gpu/src/kernel_selector/cl_kernels/reduce_gpu_b_fs_yx_fsv16.cl]: looks good. applied

**Comment by byungilm** [comment]
Passed perf tests

**Review by isanghao — APPROVED** [review]
